### PR TITLE
Fix: Allow locking on object specified attributes, performance enhancement (fixes #183)

### LINKED ISSUE #183: set locked attribute loophole
A locked attribute can be overriden if it is set within an object as this doesn't pass through the locking checks:
https://github.com/adaptlearning/adapt-contrib-core/blob/master/js/models/lockingModel.js#L4

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes #183 

### Fix
* Do not attempt to update and trigger attributes which haven't changed
* Update attributes in batches rather than individually
* Allow locking attributes to be changed via object syntax

### CODE REVIEW

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by ethan-lp — APPROVED** [review]

File: js/models/lockingModel.js [hunk 1]
```diff
@@ -1,33 +1,55 @@
...
+      }
+
+      const isSettingValueForSpecificPlugin = options?.pluginName;
+      if (!isSettingValueForSpecificPlugin) {
+        console.error('Must supply a pluginName to change a locked attribute');
+        options.pluginName = 'compatibility';
+      }
+
+      const pluginName = options.pluginName;
+      if (this.defaults[attrName] !== undefined) {
```
Comment (chris-steele) [inline js/models/lockingModel.js:29]: I feel like I have missed something, but `defaults` are generally provided via a function so presumably array notation will not work as intended?

**Review by chris-steele — COMMENTED** [review]
Looking at [this line](https://github.com/adaptlearning/adapt-contrib-core/pull/256/files#diff-828c154036fceb2feb0ed7365f50e3a968b9b30923ce48e77993eec7a3d1fccdL55), where is `this.lockedAttributes` defined?

File: js/models/lockingModel.js (hunk 1 above)
Comment (oliverfoster) [reply js/models/lockingModel.js:29]: I agree. Will change. Ta.

**Comment by oliverfoster** [comment]
> Looking at [this line](https://github.com/adaptlearning/adapt-contrib-core/blob/master/js/models/lockingModel.js#L55), where is `this.lockedAttributes` defined?

https://github.com/adaptlearning/adapt-contrib-core/search?q=lockedAttributes

**Review by eleanor-heath — APPROVED** [review]

**Pull request merged**